### DISCUSSION TITLE
Error "ValueError: columns cannot be a set" from pyemu.ParameterEnsemble.from_parfiles()

### DISCUSSION THREAD

**Discussion by sjepsen395** [pypest/pyemu#523] [Q&A] [answered]
Hi All

Does anyone know how to get around the error "ValueError: columns cannot be a set" when calling pyemu.ParameterEnsemble.from_parfiles(pth_pst, parfile_names)? Input argument pth_pst = path to PEST control file, parfile_names = list of parameter value file paths. My parameter value files don't contain all parameters in my PCF, only the adjustable ones (i.e., not the fixed or tied parameters). Is this problematic for the method? PEST++ IES does not seem to mind as long as the adjustable parameters are in there, so I'm not sure why there would be a problem.

Thanks,
Steve

**Comment by briochh** [comment, marked as answer]
Hi @sjepsen395, it does look like we have a little bug in that method -- potentially not that well tested. I'll get a fix sorted asap.

**Reply by sjepsen395** [reply]
Thanks @briochh! Really appreciate it.

Steve

**Reply by sjepsen395** [reply]
Hi Brioch,

On a related note, there may be an issue with pyemu.pst.parrep on a .csv file when parameter ies_csv_by_reals is set to false. For now I am setting this parameter to true (default). One is the transpose of the other. Let me know if I should post this in a new topic. 

Thanks,
Steve

**Reply by briochh** [reply]
Sorry for the delay. If you are able to pull together simple example on this second issue then by all means post in [issues](https://github.com/pypest/pyemu/issues). There is every chance that the parrep is expecting that csv file to be a particular orientation.

**Reply by sjepsen395** [reply]
Hi Briochh,

OK. I was able to use pyemu.pst.parrep fine on an ensemble file when parameter ies_csv_by_reals was left at TRUE (default). So I just went with that since transposing was not an necessity.

It looks like issue "parrep bug? #504" may be addressing this apparent problem with parrep'ing from a transposed ensemble file.

Thanks,
Steve
